pkg/kv/iterators_test.go

```diff
@@ -0,0 +1,377 @@
+package kv_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/kv/kvtest"
+)
+
+const (
+	firstPartitionKey  = "m"
+	secondPartitionKey = "ma"
+	thirdPartitionKey  = "mb"
+)
+
+func TestIterators(t *testing.T) {
+	ctx := context.Background()
+	store := kvtest.GetStore(ctx, t)
```

Comment:
Can we add other store drivers too?
The iterators rely heavily on the different `Scan` and `Next` implementations by the different stores
In fact, the panic we fixed in #4108 is a result of the way the postgres driver implements `Scan` (which is pretty similar to dynamodb, but totally different from the mem kv)

Reply:
done